pyvista/core/dataobject.py

```diff
@@ -535,28 +535,82 @@ def copy_attributes(self, dataset: _vtk.vtkDataSet):
         self.CopyAttributes(dataset)
 
     def __getstate__(self):
-        """Support pickle. Serialize the VTK object to ASCII string."""
+        """Support pickle. Serialize the VTK object to `bytes` or `str`."""
         state = self.__dict__.copy()
-        writer = _vtk.vtkDataSetWriter()
-        writer.SetInputDataObject(self)
-        writer.SetWriteToOutputString(True)
-        writer.SetFileTypeToBinary()
-        writer.Write()
-        to_serialize = writer.GetOutputStdString()
+
+        if pyvista.PICKLE_FORMAT.upper() == 'XML':
+            if isinstance(self, _vtk.vtkImageData):
+                writer = _vtk.vtkXMLImageDataWriter()
+            elif isinstance(self, _vtk.vtkStructuredGrid):
+                writer = _vtk.vtkXMLStructuredGridWriter()
+            elif isinstance(self, _vtk.vtkRectilinearGrid):
+                writer = _vtk.vtkXMLRectilinearGridWriter()
+            elif isinstance(self, _vtk.vtkUnstructuredGrid):
+                writer = _vtk.vtkXMLUnstructuredGridWriter()
+            elif isinstance(self, _vtk.vtkPolyData):
+                writer = _vtk.vtkXMLPolyDataWriter()
+            elif isinstance(self, _vtk.vtkTable):
+                writer = _vtk.vtkXMLTableWriter()
```

Comment:
I guess we could make this slightly more concise by defining a pair of classes and writers, something like
```py
writers = {
    _vtk.vtkImageData: _vtk.vtkXMLImageDataWriter,
     _vtk.vtkStructuredGrid: _vtk.vtkXMLStructuredGridWriter,
    ...
}
for parent_type, writer_type in writers.items():
    if isinstance(self, parent_type):
        writer = writers[parent_type]
        break
else:
    raise TypeError(f'Invalid mesh type {type(self).__name__}')
```
So basically a dispatch dict with the additional step of allowing subclasses.